### ISSUE TITLE
new trainning  on meltemi 21_9

### ISSUE THREAD

**Issue by menelaosas** [menelaosas/opsyedai_chatbot_Thesis_Demokritos#10]
I am making a new training on Meltemi with dataset v5.jsonl, but I had memory allocation problems. These changes were made 
Key Changes Made to Fix the Memory Issue
1. Reduced Sequence Length
pythonMAX_LENGTH = 512  # Reduced from 1024
Why: Memory usage scales quadratically with sequence length due to attention mechanisms. Cutting from 1024→512 reduces memory by ~75%.
2. Enhanced Memory Management
pythonos.environ["PYTORCH_CUDA_ALLOC_CONF"] = "max_split_size_mb:256,expandable_segments:True"
Why: Reduces memory fragmentation and allows better memory reuse.
3. Optimized Training Parameters

Gradient accumulation: Increased from 4→8 steps (maintains effective batch size while using less memory per step)
Epochs: Reduced from 10→5 (faster training, less memory pressure over time)
Save limit: Reduced checkpoints to save disk space

4. Memory Monitoring Functions
Added clear_gpu_memory() and get_gpu_memory_info() to track and manage memory usage throughout training.

run
complete_optimized_training.py
dataset_v5.jsonl

- [x] after training check model
- [ ] I kept a photo from GPU usage, put it in the thesis
- [ ] keep the testing results in an Excel file to Visualization them
- [ ] test model accuracy
- [ ] keep results to Visualization them
- [x] not successful in proceeding
